docs/en_US/guide/ai/python_openvino_tutorial.md

```diff
@@ -0,0 +1,285 @@
+
+    def is_aggregate(self):
+        return False
+
+
+inferenceIns = InferenceFunc()
+```
+
+Once the code is implemented, we also need to add a description file for each function, which is placed in the functions directory, in this case we create the `defect.json` file. This file will help the eKuiper manager to automatically produce the UI associated with the plugin.
+
+### Plugin Packaging
+
+At this point, we have completed the development of the main functionality, and next we need to package these files into a plugin format. Plugin packaging requires several steps to be completed.
+
+1. If the plugin has additional dependencies, you need to create the dependency installation script `install.sh`. When the plugin is installed, eKuiper will look for an installation script file `install.sh` in the plugin package and execute it if there is one. In this case, we create a `requirements.txt` file listing all the dependency packages. The installation of the dependencies is done in `install.sh` by calling `pip install -r $cur/requirements.txt`. For other plugins, you can reuse this script to update `requirements.txt` if you have no special requirements.
+2. Create a Python entry file that exposes all the implemented interfaces. Because multiple extensions can be implemented in a single plugin, you need an entry file that defines the implementation classes for each extension. The content is a main function, which is the entry point for the plugin runtime. It calls the methods in the SDK to define the plugin, including the plugin name, and a list of keys for the source, sink, and function implemented in the plugin. Here only a function plugin named `inference` is implemented, and its corresponding implementation method is `inferenceIns`. The Python plug-in process is independent of the eKuiper main process.
+
+    ```python
+    if __name__ == '__main__':
+        # Define the plugin
+        c = PluginConfig("defect", {}, {},
+                         {"inference": lambda: inferenceIns})
+        # Start the plugin instance
+        plugin.start(c)
+    ```
+
+3. Create a plugin description file in JSON format to define the metadata of the plugin. The file name must be the same as the plugin name, i.e. `defect.json`. The function names defined therein must correspond exactly to the entry file, and the contents of the file are as follows. Where, executable is used to define the name of the plugin's executable entry file.
+
+    ```json
+    {
+      "version": "v1.0.0",
+      "language": "python",
+      "executable": "main.py",
+      "sources": [
+      ],
+      "sinks": [
+      ],
+      "functions": [
+        "inference"
+      ]
+    }
+    ```
+
+At this point we have completed the development of the plugin, next we just need to package all the files in the directory into a zip file. The zip file should have a file structure similar to:
+
+- inference.py
+- requirements.txt
+- install.sh
+- main.py
+- defect.json
+- models
+  - model.bin
```

Comment:
Are these models public? Can we disclose them? Can we just put a link here instead of ship them?

Reply:
Intel do not have  a public link that holds these models